Clang compilation fix

GeorgeIvlev opened the pull request:
- Resolve warnings related to bad comment string ( /** ... /**/ )
- Fix logical warnings/errors on clang compilation
- Fix critical code errors to be able to build project with clang
- Fix critical errors in code
- Update main CMakeLists file for better development experience
   - Added output for libraries to one place after build.
   - Group part of unix configuration into one block
- Resolved couple of problems that was found in previous commits of this fix.

eagleivg committed:
xrEngine: hide some SDL2 flags under version define (fix build on Ubuntu16.04 with SDL2 v2.0.0)

eagleivg committed:
Common: implement file & directory remove on Linux

eagleivg committed:
xrCore: added getting date of files under Linux/FreeBSD

GeorgeIvlev committed:
- Resolve warnings related to bad comment string ( /** ... /**/ ).

GeorgeIvlev committed:
- fix logical warnings/errors for clang compilation.

GeorgeIvlev committed:
- fix critical code errors to be able to build project with clang.

GeorgeIvlev committed:
- fix critical errors in code.

TODO: Need to figure out better solution for that fixes.

GeorgeIvlev committed:
- Update main CMakeLists file for better development experience.
 - Added output for libraries to one place after build.
 - Group part of unix configuration into one block.

GeorgeIvlev committed:
- Resolved couple of problems that was found earlier.

Xottab-DUTY commented on src/xrSound/OpenALDeviceList.cpp:
```diff
@@ -199,8 +199,8 @@ void ALDeviceList::SelectBestDevice()
-            if (majorVersion > best_majorVersion ||
-                majorVersion == best_majorVersion && minorVersion > best_minorVersion)
+            if ((majorVersion > best_majorVersion) ||
+                ((majorVersion == best_majorVersion) && minorVersion) > best_minorVersion)
```
Wrong parentheses.

GeorgeIvlev replied:
You mean that its should be like that?
```c++
if ((majorVersion > best_majorVersion) ||
    ->( ((majorVersion == best_majorVersion) && minorVersion) > best_minorVersion) )*<-*
```

Xottab-DUTY replied:
Clang is complaining about this?
If true, then is should be:
```cpp
if (majorVersion > best_majorVersion ||
	(majorVersion == best_majorVersion && minorVersion > best_minorVersion))
```
If no warnings, then we don't need to change the code at all.

GeorgeIvlev replied:
Yes, it throws error without using parentheses.

GeorgeIvlev replied:
I have removed non-necessary brackets.

Xottab-DUTY commented on src/xrUICore/ListWnd/UIListWnd_inline.h:
```diff
@@ -16,10 +16,9 @@ bool CUIListWnd::AddItem(const char* str, const float shift, void* pData,
-                                ? GetHeight() - GetSize() * m_iItemHeight - m_iItemHeight
-                                : GetSize() * m_iItemHeight,
-                m_iItemWidth, m_iItemHeight);
+    pItem->Init(str, shift, m_bVertFlip ? GetHeight() - GetItemsCount() * m_iItemHeight - m_iItemHeight
```
Don't change the formatting.

GeorgeIvlev replied:
Ok, will update.

GeorgeIvlev commented on src/Layers/xrRender/r__screenshot.cpp:
```diff
@@ -59,8 +59,12 @@ bool CreateImage(fipMemoryIO& output, FREE_IMAGE_FORMAT format, u8*& buffer, DWO
+        (u8)(width / 256),
+        (u8)(height % 256),
+        (u8)(height / 256),
         bits, 0
```
I think we can update that code by change divide operation to bit shift:
```
(u8)(width / 256) -> (u8)(width >> 8)
(u8)(height / 256) -> (u8)(height >> 8)
```
Because divide by 256 equals to bit shift to right by 8 bits (2^8 = 256) but without float precision.

GeorgeIvlev committed:
- Formating reverted back.

GeorgeIvlev committed:
- Remove non-necessary parentheses.

GeorgeIvlev committed:
- fix problem with file not found on MVS compilation ( Windows ).

GeorgeIvlev committed:
- Format code. Simplify parentheses.

Xottab-DUTY commented on src/xrGame/game_sv_item_respawner.cpp:
```diff
@@ -271,7 +271,7 @@ void item_respawn_manager::add_new_rpoint(shared_str profile_sect, RPoint const&
         {
 #ifndef MASTER_GOLD
-            Msg("! ERROR: failed to create entity [%s] with addons [%d]", iter_rsect->section_name, iter_rsect->addons);
+  Msg("! ERROR: failed to create entity [%s] with addons [%d]", iter_rsect->section_name.c_str(), iter_rsect->addons);
```
```suggestion
            Msg("! ERROR: failed to create entity [%s] with addons [%d]", iter_rsect->section_name.c_str(), iter_rsect->addons);
```

GeorgeIvlev committed:
Update src/xrGame/game_sv_item_respawner.cpp

- Restore formatting.

Co-Authored-By: Xottab_DUTY <Xottab-DUTY@users.noreply.github.com>

GeorgeIvlev committed:
Merge branch 'xd_dev-openxray' into xd_dev

GeorgeIvlev committed:
Merge branch 'xd_dev' into fix/clang_compilation

# Conflicts:
#	src/Layers/xrRenderPC_GL/r2_R_sun.cpp

Xottab-DUTY committed:
Fixed some not working smart covers in CS (#382)

Xottab-DUTY committed:
Fixed Linux compilation

Xottab-DUTY committed:
Removed SDL_WINDOW_SKIP_TASKBAR flag from splash screen

It doesn't work on SDL 2.0.5 on Windows anyway and I find it fancy when splash screen is added to the taskbar.

Xottab-DUTY committed:
Fixed wrong vsprintf_s usage and linux compilation

Xottab-DUTY committed:
Fixed Linux compilation (honestly)

Xottab-DUTY committed:
Forgot to tell vsprintf_s about size...

Xottab-DUTY committed:
xrRenderDX10/Blender_Recorder_R3.cpp: optimized sampler name traversal

Xottab-DUTY committed:
xrRender_R*/r*_shaders.cpp: fixed possible crash when failed to disassmble shader

Xottab-DUTY committed:
xrRender_R4/r4_shader.cpp: cleanup

Xottab-DUTY committed:
xrRender_R4: added support for non-unique named shader resources. (Closes #281)

Xottab-DUTY committed:
xrRender: resource managers and constants: use modern emplace_back and emplace

Xottab-DUTY committed:
xrRender/ResourceManager.cpp: use range-based for loop

Also nullptr and some formatting.

vTurbine committed:
xrCore: application title member added

Xottab-DUTY committed:
xrCore: application title member added (#484)

xrCore: application title member added

Xottab-DUTY committed:
Prevent string concatenation crash on MASTER_GOLD

Matches original engine behaviour.

Xottab-DUTY committed:
Fixed crash at game start after ded6f1842e97f29ef135ed706f3aa06a6e1e6a0d
And fixed other possible issues

Xottab-DUTY committed:
Fixed elite detector doesn't show artefacts (#382)

vTurbine committed:
xrRenderer: common types introduced

Xottab-DUTY committed:
Merge pull request #485 from vTurbine/renderer/common_types

xrRenderer: common types introduced

Xottab-DUTY committed:
[skip ci] Updated Clear Sky status

Xottab-DUTY committed:
Second fix to bloodsucker drag & jump in CS (#382)

Addition to 610cac6abcfab4cc3ca8cfbc9acaa6b243891702

Xottab-DUTY committed:
Set different window title depending on game mode

Xottab-DUTY committed:
xr_3da/resource.rc: removed unused startup dialog

Xottab-DUTY committed:
xr_3da/resource.rc: convert to UTF-8, fixed wrong Polish language identifiers, resort

Xottab-DUTY committed:
xr_3da/resource.rc: resort and regeneration

Xottab-DUTY committed:
xr_3da: added Clear Sky and Shadow of Chernobyl icons

Xottab-DUTY committed:
Set Clear Sky or Shadow of Chernobyl application icon if in the corresponding mode
Refactored embedded resources management

Xottab-DUTY committed:
More correct way to fix DX9 fullscreen

Rework of 32e64f5ff09afd02de366de9c30d408903dbb832

Xottab-DUTY committed:
xrEngine/Device_create.cpp: made drawBorders static bool

No need to always perform strstr, because the result won't change until
next launch.

Xottab-DUTY committed:
xrEngine/Device_create.cpp: replaced SDL_FALSE with SDL_DISABLE to suppress warning about enum...

Xottab-DUTY committed:
Always switch display mode to fullscreen when it's selected (even for DX9 renderer)
Because game looks ugly when using fullscreen desktop for DX9.
But DX9 itself will still behave as windowed.

Xottab-DUTY committed:
Always name threads created by SDL

Xottab-DUTY committed:
[skip ci] xrEngine/Device_initialize.cpp: just minor code swap
